cmd/nknc/commands/asset.go

```diff
@@ -0,0 +1,216 @@
+package commands
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+
+	"github.com/howeyc/gopass"
+	api "github.com/nknorg/nkn/v2/api/common"
+	"github.com/nknorg/nkn/v2/api/httpjson/client"
+	"github.com/nknorg/nkn/v2/common"
+	"github.com/nknorg/nkn/v2/config"
+	"github.com/nknorg/nkn/v2/util/log"
+	"github.com/nknorg/nkn/v2/vault"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	RANDBYTELEN = 4
+)
+
+// assetCmd represents the asset command
+var assetCmd = &cobra.Command{
+	Use:   "asset",
+	Short: "asset registration, issuance and transfer",
+	Long:  "",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		showusage := true
+		cmd.Flags().Visit(func(name *pflag.Flag) {
+			showusage = false
+		})
+		if showusage {
+			return cmd.Usage()
+		}
+		if err := assetAction(); err != nil {
+			log.Error(err)
+		}
+		return nil
+	},
+}
+
+var (
+	issue     bool
+	transfer  bool
+	to        string
+	value     string
+	fee       string
+	assetname string
+	symbol    string
+	precision uint32
+)
+
+func init() {
+	rootCmd.AddCommand(assetCmd)
+
+	assetCmd.Flags().BoolVarP(&issue, "issue", "i", false, "issue asset")
+	assetCmd.Flags().BoolVarP(&transfer, "transfer", "t", false, "transfer asset")
+	assetCmd.Flags().StringVarP(&WalletFile, "wallet", "w", config.Parameters.WalletFile, "wallet name")
+	assetCmd.Flags().StringVar(&WalletPassword, "password", "", "wallet password")
+	assetCmd.Flags().StringVar(&to, "to", "", "asset to whom")
+	assetCmd.Flags().StringVarP(&value, "value", "v", "", "asset amount in transfer asset or totalSupply in inssue assset")
+	assetCmd.Flags().StringVarP(&fee, "fee", "f", "", "transaction fee")
+	assetCmd.Flags().Uint64Var(&Nonce, "nonce", 0, "nonce")
+	assetCmd.Flags().StringVar(&assetname, "name", "", "asset name")
+	assetCmd.Flags().StringVar(&symbol, "symbol", "", "asset symbol")
+	assetCmd.Flags().Uint32Var(&precision, "precision", 0, "asset precision")
+
+	assetCmd.Flags().MarkHidden("password")
+	assetCmd.MarkFlagRequired("wallet")
+	assetCmd.MarkFlagRequired("password")
```

Comment:
I think these two flags should not mark required since we allow from `default value in config` and from `gopass.GetPasswd`.

Reply:
fixed.